domain/doc.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package domain contains the implementation of the domain services.
+// Each domain service is responsible for providing apis pertaining
+// to a logical domain. Domains may cross entity boundaries; the
+// partitioning is done based on packaging cohesive functional behaviour
+// not individual entity concerns.
+//
+// # Domain services
+//
+// Each domain service package has several key artefacts:
+//   - the service providing public APIS called by api server facades.
+//   - the state providing functionality to read and write persistent data.
+//   - params structs which are used as arguments / results for service API calls.
+//   - arg structs which are used as arguments / results for state calls.
+//   - type structs which are used a sqlair in/out parameters.
+//
+// The layout of a domain package providing a service is as follows:
+//
+//		domain/foo/
+//		 |- entities.go [1]
+//		 |- types.go [2]
+//		 |- params.go [3]
+//		 |- service/
+//		   |- service.go
```

Comment:
There should be a note on what tests are expected in service vs state.

Reply:
added some initial content